resources/js/components/SignIn.js

```diff
@@ -1,292 +1,250 @@
-import { Link as LinkRouter, useNavigate } from "react-router-dom";
+import React, { useEffect } from 'react';
+import { connect } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
 
-import Avatar from "@mui/material/Avatar";
-import Button from "@mui/material/Button";
-import TextField from "@mui/material/TextField";
-import FormControlLabel from "@mui/material/FormControlLabel";
-import Checkbox from "@mui/material/Checkbox";
-import Link from "@mui/material/Link";
-import Paper from "@mui/material/Paper";
-import Box from "@mui/material/Box";
-import Grid from "@mui/material/Grid";
-import Typography from "@mui/material/Typography";
-import Dialog from "@mui/material/Dialog";
-import DialogTitle from "@mui/material/DialogTitle";
-import CircularProgress from "@mui/material/CircularProgress";
-import IconButton from "@mui/material/IconButton";
-import InputAdornment from "@mui/material/InputAdornment";
-import Visibility from "@mui/icons-material/Visibility";
-import VisibilityOff from "@mui/icons-material/VisibilityOff";
+import Avatar from '@mui/material/Avatar';
+import Button from '@mui/material/Button';
+import TextField from '@mui/material/TextField';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import Checkbox from '@mui/material/Checkbox';
+import Link from '@mui/material/Link';
+import Paper from '@mui/material/Paper';
+import Box from '@mui/material/Box';
+import Grid from '@mui/material/Grid';
+import Typography from '@mui/material/Typography';
+import Dialog from '@mui/material/Dialog';
+import DialogTitle from '@mui/material/DialogTitle';
+import CircularProgress from '@mui/material/CircularProgress';
+import IconButton from '@mui/material/IconButton';
+import InputAdornment from '@mui/material/InputAdornment';
+import Visibility from '@mui/icons-material/Visibility';
+import VisibilityOff from '@mui/icons-material/VisibilityOff';
 
-import Footer from "./layouts/Footer";
+import Footer from './layouts/Footer';
 import {
-    freshState,
-    signIn,
-    validateEmail,
-    validatePassword,
-    showPassword,
-    disableSubmit,
-} from "../actions/userAuthActions";
+  freshState,
+  signIn,
+  validateEmail,
+  validatePassword,
+  showPassword,
+  disableSubmit
+} from '../actions/userAuthActions';
 
 const SignIn = (props) => {
-    const navigate = useNavigate();
+  const navigate = useNavigate();
 
-    // When navigated to this component, re-initialize the component state
-    useEffect(() => {
-        props.freshState();
-        /* 
-            When navigated to this component, the password should be cleared
-            but when you put password in this component and try to navigate to
-            Sign Up, the password field has values. This checks if password has
-            values on first render then clear the values.
-        */
-    }, []);
+  // Check requestError, if false then navigate to dashboard page
+  useEffect(() => {
+    if (props.requestError === false) {
```

Comment:
no need to explicitly add `=== false` if `props.requestError`'s value is boolean, you can just do `if (props.requestError)`

Reply:
When removing the `=== false`  makes the redirection after successful request don't work.

Reply:
@michaelsilverio okay disregard the comment na lang.